sw/airborne/arch/chibios/mcu_periph/i2c_arch.c

```diff
@@ -198,7 +198,7 @@ static struct i2c_init i2c1_init_s = {
 struct i2c_errors i2c1_errors;
 // Thread
 static __attribute__((noreturn)) void thd_i2c1(void *arg);
-static THD_WORKING_AREA(wa_thd_i2c1, 1024);
+static THD_WORKING_AREA(wa_thd_i2c1, 128);
```

Comment:
I would be worried that this might overflow for larger i2c buffers - could you add `ifndef` guard and specify this number in the board file to override the default?

Reply:
I don't think that's how it works. The data goes in different buffers. E.g. the ublox I2C gps (used in this PR) actually is a rather large buffer user compared to al other sensors, and it does not seem to be a problem. Also, the chibios examples themselves use this number. I think they were oversized in pprz. Of course could still add the ifdef overrides if still required, but think these numbers here should be the default.

I'm not sure though what happens when the buffer is to small, I may try that when I get the chance, 
It would also be nice if someone with e.g. a Chimera can test this.

(same for all the other THD_WORKING_AREA's)

Reply:
OK, sounds reasonable. I think ChibiStudio plugin for Eclipse can tell you the used memory for each thread.

Reply:
i2c buffers are global variables so not counted for the stack size. Only automatic variables (more or less the local variables) created by the thread will use the stack memory, a few pointers in the case of this driver. So I guess 128 is fine.